### PR TITLE
Circle ci fixes automated build

### PR DESCRIPTION
**Pull request by srclosson**
Why?

In the previous version, only a console error log was done to indicate failure, there was no error thrown. As a result, all publishes are "successful". Fixed.

Make the public "smarter". If an environment variable for GITHUB_TOKEN is already used, detect and set appropriately.

Run in linked mode ONLY if there is a link created for toolkit AND toolkit is a symbolic link.

Basic testing infrastructure.

### CODE REVIEW

File: packages/grafana-toolkit/src/cli/utils/githubRelease.ts [hunk 1]
```diff
@@ -97,7 +98,13 @@ class GitHubRelease {
         `https://uploads.github.com/repos/${this.username}/${this.repository}/releases/${newReleaseResponse.data.id}/assets`
       );
     } catch (reason) {
-      console.error('error', reason);
+      if (reason.data.message) {
+        console.error(reason.data.message);
+      } else {
+        console.error(reason);
+      }
```
Comment (stevenvachon) [inline packages/grafana-toolkit/src/cli/utils/githubRelease.ts:105]: You might prefer
```js
console.error(reason.data.message ?? reason);
// or?
console.error(reason.data?.message ?? reason);
```

File: packages/grafana-toolkit/src/cli/tasks/plugin.utils.ts [hunk 2]
```diff
@@ -162,13 +162,24 @@ const githubPublishRunner: TaskRunner<GithubPublishOptions> = async ({ dryrun, v
...
 
-  if (!process.env['GITHUB_TOKEN']) {
-    throw `Github publish requires that you set the environment variable GITHUB_TOKEN to a valid github api token.
-    See: https://github.com/settings/tokens for more details.`;
+  if (!process.env['GITHUB_ACCESS_TOKEN']) {
+    // Try to use GITHUB_TOKEN, which may be set.
+    if (process.env['GITHUB_TOKEN']) {
+      process.env['GITHUB_ACCESS_TOKEN'] = process.env['GITHUB_TOKEN'];
+    } else {
+      throw `Github publish requires that you set the environment variable GITHUB_ACCESS_TOKEN to a valid github api token.
```
Comment (stevenvachon) [inline packages/grafana-toolkit/src/cli/tasks/plugin.utils.ts:170]: We should always throw instances of `Error`. I don't know how much digging may be necessary to fix that in this PR, though.

**Review by ryantxu — APPROVED** [review]
Looks good

**Review by stevenvachon — APPROVED** [review]

File: packages/grafana-toolkit/src/cli/utils/githubRelease.ts (hunk 1 above)
Comment (srclosson) [reply packages/grafana-toolkit/src/cli/utils/githubRelease.ts:105]: Yes, I do actually :)

**Pull request merged**